test/downgrade/knative_postdowngrade_test.go

```diff
@@ -0,0 +1,127 @@
+	t.Run("verify resources", func(t *testing.T) {
+		resources.AssertKEOperatorCRReadyStatus(t, clients, names)
+		resources.SetKodataDir()
+		defer os.Unsetenv(common.KoEnvKey)
+
+		targetManifest, err := common.TargetManifest(&v1alpha1.KnativeEventing{})
+		if err != nil {
+			t.Fatalf("Failed to get the manifest for Knative: %v", err)
+		}
+
+		instance := &v1alpha1.KnativeEventing{
+			Spec: v1alpha1.KnativeEventingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Version: test.OperatorFlags.PreviousEventingVersion,
+				},
+			},
+		}
+
+		// Based on the previous release version, get the deployment resources.
+		preManifest, err := common.TargetManifest(instance)
+		if err != nil {
+			t.Fatalf("Failed to get KnativeEventing manifest: %v", err)
+		}
+		expectedDeployments := resources.GetExpectedDeployments(preManifest)
+		util.AssertEqual(t, len(expectedDeployments) > 0, true)
+		resources.AssertKnativeDeploymentStatus(t, clients, names.Namespace, test.OperatorFlags.PreviousEventingVersion,
+			expectedDeployments)
+		resources.AssertKnativeObsoleteResource(t, clients, names.Namespace,
+			targetManifest.Filter(mf.Not(mf.In(preManifest))).Resources())
+	})
+}
+
+// TestKnativeServingPostDowngrade verifies the KnativeServing creation, after downgraded to the previous version.
+func TestKnativeServingPostDowngrade(t *testing.T) {
+	clients := client.Setup(t)
+
+	names := test.ResourceNames{
+		KnativeServing: test.OperatorName,
+		Namespace:      test.ServingOperatorNamespace,
+	}
+
+	// Create a KnativeServing
+	if _, err := resources.EnsureKnativeServingExists(clients.KnativeServing(), names); err != nil {
+		t.Fatalf("KnativeService %q failed to create: %v", names.KnativeServing, err)
+	}
+
+	// Verify if resources match the requirement for the previous release after downgrade
+	t.Run("verify resources", func(t *testing.T) {
+		resources.AssertKSOperatorCRReadyStatus(t, clients, names)
+		resources.SetKodataDir()
+		defer os.Unsetenv(common.KoEnvKey)
+
+		targetManifest, err := common.TargetManifest(&v1alpha1.KnativeServing{})
+		if err != nil {
+			t.Fatalf("Failed to get the manifest for Knative: %v", err)
+		}
+
+		instance := &v1alpha1.KnativeServing{
+			Spec: v1alpha1.KnativeServingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Version: test.OperatorFlags.PreviousServingVersion,
+				},
+			},
+		}
+
+		// Based on the previous release version, get the deployment resources.
+		preManifest, err := common.TargetManifest(instance)
+		if err != nil {
+			t.Fatalf("Failed to get KnativeServing manifest: %v", err)
+		}
+		expectedDeployments := resources.GetExpectedDeployments(preManifest)
+		util.AssertEqual(t, len(expectedDeployments) > 0, true)
+		resources.AssertKnativeDeploymentStatus(t, clients, names.Namespace, test.OperatorFlags.PreviousServingVersion,
+			expectedDeployments)
+		resources.AssertKnativeObsoleteResource(t, clients, names.Namespace,
+			targetManifest.Filter(mf.Not(mf.In(preManifest))).Resources())
```

Comment:
Is there a way for us to make a 'fake' downgrade for tests? This relies on the downgrade being 'correct' which seems fragile depending on how the operand implements things.

Don't feel too strongly about it though, as I think this is a good qualification improvement.

/lgtm